arangod/ClusterEngine/ClusterIndex.cpp

```diff
@@ -123,6 +130,57 @@ void ClusterIndex::toVelocyPack(
     builder.add(StaticStrings::IndexEstimates, VPackValue(false));
   }
 
+  if (Index::hasFlag(flags, Index::Serialize::Figures)) {
+    uint64_t progress = 0;
+    uint64_t success = 0;
+    auto const shards = _collection.shardIds();
+    auto const body = VPackBuffer<uint8_t>();
+    auto* pool =
+        _collection.vocbase().server().getFeature<NetworkFeature>().pool();
+    std::vector<Future<network::Response>> futures;
+    futures.reserve(shards->size());
+    std::string const prefix = "/_api/index/";
+    for (auto const& shard : *shards) {
+      network::RequestOptions reqOpts;
+      std::string const url =
+          prefix + shard.first + "/" + std::to_string(_iid.id());
+      futures.emplace_back(
+          network::sendRequestRetry(pool, "shard:" + shard.first,
```

Comment:
Is this code can be called for index which is already finished creation?
If yes I think it should be fixed.

Reply:
Again, only, when the figures are explicitly requested. This is a very verbose output. Due to time constraints, I would suggest a fix after my holidays.